systemd/verdaccio.service

```diff
@@ -1,12 +1,12 @@
 [Unit]
-Description=verdaccio Service
+Description=Verdaccio lightweight npm proxy registry
 
 [Service]
 Type=simple
+Restart=always
 User=verdaccio
-WorkingDirectory=/home/verdaccio
-ExecStart=/usr/bin/env verdaccio
-ExecStop=/usr/bin/bash -c "kill $(ps -ef | grep ^verdacc+ | awk {'print $2'})"
+Group=verdaccio
+ExecStart=/usr/bin/verdaccio --config /etc/verdaccio/config.yaml
```

Comment:
The default conf is not there by default, how the user knows about this location?

Reply:
`/home/verdaccio` is not there by default neither :)
The cause of this pull request is to make systemd unit production-ready, e.g. make it usable for system-wide installation on production servers (as opposed to development or home servers).
As FHS (Filesystem Hierarchy Standard, `man 7 hier`) states, variable state information for programs (e.g. data) should be stored in `/var/lib/` (e.g. `/var/lib/verdaccio` in your case), while `/home/` is for home directories of users (human users, not system). Moreover, system users usually do not have home directories (of course they have some directory specified as their home directory, but no "personal" files are stored there). Thus using `/home/verdaccio` for data storage is not correct in terms of FHS and that's why I removed `WorkingDirectory=/home/verdaccio`.
As for configuration files, FHS states that they should be stored in `/etc` (so in your case that would be `/etc/verdaccio.yaml` or `/etc/verdaccio/*.yaml`). None of example configuration files would work out of the box on production (`default.yaml` - "It allows all users to do anything, so don't use it on production systems.", `docker.yaml` - uses non-standard paths, `full.yaml` - has a lot of placeholders to change), so anyway user would have to create his own copy of configuration file, so I think `/etc/verdaccio/config.yaml` would be an apparent name for a arbitrary system-wide config for Verdaccio.

Reply:
Do we have some kind of package build automation? (e.g. deb or rpm) Or is anybody downstream building and packaging this for linux distributions?

In any case, I agree with @tiandrey about the comments, if daemonizing verdaccio, as a sysadmin that's where I would expect to find the configuration files and runtime state data.

Reply:
I personally use fpm (https://github.com/jordansissel/fpm) - it builds DEBs and RPMs from npm just fine. And it automatically installs configs from `conf` subdir to `/etc/verdaccio`.